frontend/docsettings.lua

```diff
@@ -345,27 +359,45 @@ end
 
 --- Updates sidecar info for file rename/copy/move/delete operations.
 function DocSettings:update(doc_path, new_doc_path, copy)
+    local doc_settings, new_sidecar_dir
+
+    -- update metadata
     if self:hasSidecarFile(doc_path) then
-        local doc_settings = DocSettings:open(doc_path)
+        doc_settings = DocSettings:open(doc_path)
```

Comment:
There's something that looks utterly messy to me around all this. I don't know how much was already there, and how much comes with this PR.
There are some DocSettings: functions that are methods (expect and apply to an expected `self`) and others that are "class methods", ie. plain functions accessed as `DocSettings:open(doc_path)`, that should not touch/use self (or that can use `self` but `self` is the class DocSettings object.

Below, you will use `local cover_file = self:getCustomBookCover(doc_path)`.
As you made it expecting a doc_path argument, it's supposed to be a class function (if it were a method of a DocSettings instances, you would use self.doc_path or something like that, ie. the DocSetting instance should know it's doc_path, and you would not need to provide it). But it will store the cached cover_file in the class object !

But we already have somewhere (in other classes and may be in here) some "hybrid" functions that can act as both instance methods and class methods.... which is always hard to grasp. So, dunno if it was already that messy in DocSettings...

Does what I say make sense to you ? Haven't you find such stuff confusing when making this PR?

Reply:
Yes, it does.
I will look at it more thoroughly.

Reply:
Thanks :)
I think the public instance method should be named :getCoverFile(), so we can have in the other modules readable `local cover_file = self.ui.doc_settings:getCoverFile()`.

Then, you can have your other "helpers" as class methods (private, or available if needed elsewhere):
```
DocSettings:getCoverFile(sidecar_dir) -- may be better named :findCoverFileInSidecarDir()
DocSettings:getCustomBookCover(doc_path) -- may be better named :findCoverFile()
```

Reply:
Do we have a naming rule to distinguish methods (like `Docsettings:purge()`) and functions/"class methods" (like `DocSettings:update(doc_path, new_doc_path, copy)`)?
I feel that `update` was a wrong name.

Reply:
I don't think we have any naming rule - and I don't remember having thought about it whenever I needed to have both instance and class methods.

There's another option we do use in some places, it's to use `Class:method()` (with `:`) for instance methods - and only `Class.function()` (with `.`) for pure functions/class methods - so the `:` / `.` obviously indicate what kind it is - even if it is a bit hard to notice :)

> I feel that update was a wrong name from that point of view.

Indeed, it sounds like it would be an instance method to "update" itself (and not clear what "update" means :))
So, with the above idea, it could be named:
`DocSettings.moveOrCopyDocSettings(doc_path, new_doc_path, copy)`